06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * Implementation of the inverted index app.
+ * @class
+ */
+class InvertedIndex {
+  /**
+   * @desc Initializes indexedFiles object
+   */
+  constructor() {
+    this.indexedFiles = {};
+  }
+
+  /**
+   * @desc readFile: Reads the contents of file as text
+   * @param {file} file - the file to be read
+   * @param {string} fileName - the name of the file to be read
+   * @return {promise} - promise object to be acted on if file
+   * reading is succesful or not
+   */
+  readFile(file, fileName) {
+    this.file = file;
+    const validJson = /.+\.json$/;
+
+    if (!validJson.exec(fileName)) {
+      return false;
+    }
+
+    return new Promise((resolve, reject) => {
+      const fileReader = new FileReader();
+
+      fileReader.onload = (fileBeingRead) => {
+        const fileContent = fileBeingRead.target.result;
+
+        try {
+          JSON.parse(fileContent);
+        } catch (err) {
+          reject('This JSON file is invalid. Check the file and try again.');
+        }
+
+        resolve(fileContent);
+      };
+
+      fileReader.readAsText(this.file);
+    });
+  }
+
+  /**
+   * @desc validateFile: Checks if documents
+   *  in object have the specified structure of title and text keys
+   * @param {Object} file
+   * - the object to validate for required structure
+   * @throws
+   * - throws an error if object doesn't does not have title and text keys
+   * @return {Object} - returns object back to caller if object is valid
+   */
+  validateFile(file) {
+    this.file = file;
+    this.file.forEach((doc) => {
+      if (!(Object.prototype.hasOwnProperty.call(doc, 'title')
+      && Object.prototype.hasOwnProperty.call(doc, 'text'))) {
+        throw new Error(`Your JSON file does not have the expected format.
+          Documents are to have title and text keys alone.`);
+      }
+    });
+
+    return this.file;
+  }
+
+  /**
+   * @desc tokenize: Returns an array of valid words in all text fields of file.
+     Also returns an array of valid words in a single string.
+   * @param {(Object | String)} file - object or string to be tokenized
+   - the object whose texts are to be tokenized or string to tokenize
+   * @return {array} - array of valid words
+   */
+  tokenize(file) {
+    const validWord = /[a-zA-Z]+/g;
+    this.file = file;
+    let words = '';
+    let found;
+    const validWordsArray = [];
+
+    if (typeof (this.file) === 'object') {
+      for (let i = 0; i < this.file.length; i += 1) {
+        words = words.concat(`${this.file[i].text} `);
```

Comment:
I am trying to build a single string out of all 'text' fields in an uploaded file, so that I can get all the words out.